### ISSUE TITLE
(minor) Peculiar placement of speech balloon tail

### ISSUE THREAD

**Issue by brianharvey** [jmoenig/Snap#2537]

**Comment by jguille2** [comment]
Hi @brianharvey ,

I don't understand this issue :(

The speech balloon is positioned without regard to the other sprites situation or its own orientation. It tries to be on the right....and when this position takes him off the stage (when it is closer to the right edge) then it turns to be on the left.

Is this the problem?

Joan

**Comment by jmoenig** [comment]
Yeah, Brian wants sprites to know about things like their "head" :-)
And, of course, speech balloons must be positionable by absolute and relative coordinates and their font, size, weight and line-breaks specifiable...

**Comment by brianharvey** [comment]
Doesn't have to be perfect.  But in this example, if the speech balloon ends up on top of the sprite, rather than off to the side, then just the tail could be smarter, pointing more down than sideways.

**Comment by DarDoro** [comment]
Talk bubble is positioned on top purposelly (???).
_**SpriteMorph.prototype.positionTalkBubble @ objects.js**_
line 5106 ^ 5115 can be modified to get more consistent bahaviour
e.g
```
bubble.setLeft(this.center().x);
...
bubble.setRight(this.center().x);

```
or
```
bubble.setLeft(this.right());
...
bubble.setRight(this.left());
```

More elaborated version can set already processed **_"isPointingRight"_** based on direction of sprite.
